src/esm_runscripts/filedicts.py

```diff
@@ -146,6 +163,22 @@ def __init__(self, full_config: dict, attrs_address: dict):
         # Checks
         self._check_path_in_computer_is_abs()
 
+    def __setattr__(self, name: str, value: Any) -> None:
+        """Checks when changing dot attributes for disallowed values"""
+        if name == "datestamp_format":
+            self._check_datestamp_format_is_allowed(value)
+        if name == "datestamp_method":
+            self._check_datestamp_method_is_allowed(value)
+        return super().__setattr__(name, value)
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        """Checks for changing with sim_file['my_key'] = 'new_value'"""
+        if key == "datestamp_format":
+            self._check_datestamp_format_is_allowed(value)
+        if key == "datestamp_method":
+            self._check_datestamp_method_is_allowed(value)
+        return super().__setitem__(key, value)
```

Comment:
This is a really good idea :D, I'll keep these methods in mind for when we cleanup, there might be some variables that we want to check here, instead of in the __init__

Reply:
Thanks! That was one of the unit tests I wrote down and thought "oh if we don't check this it will probably bite us in the butt somehow"